src/connector/connector.ts

```diff
@@ -83,6 +101,11 @@ export abstract class Connector {
      */
     abstract connect(): void;
 
+    /**
+     * Listen for an event on a channel instance.
+     */
+    abstract listen(name: string, event: string, callback: CallableFunction): Channel;
```

Comment:
This seems to have been present in all supplied connectors and this would be a basic requirement/naming for any different provider's connector.